### PR TITLE
bmt, file: Extract AsyncHasher to separate package

### PR DESCRIPTION
**Pull request by nolash**
This PR is part of a series of PRs that introduces an interface that allows chaining of components that receive a data stream and generate hashes and intermediate Merkle-Tree chunks. The individual PR steps will be partitioned from https://github.com/ethersphere/swarm/pull/2022 (branch https://github.com/nolash/swarm/tree/filehasher-avenged ) as follows:

1. ~~Introduce `SectionWriter`,  implement this interface in `bmt`, make `AsyncHasher` standalone~~
1. **Move `AsyncHasher` to `file/hasher` (this PR)** 
1. Add reference implementation of the Filehasher algorithm
1. Add implementation of `SectionWriter` sub-component for hashing intermediate Merkle Tree levels.
1. Add implementation of `SectionWriter` component executing the FileHasher algorithm 
1. Add a "splitter" that bridges `io.Reader` and `SectionWriter`, and an implementation of `SectionWriter` component that provides `Chunk` output.
1. Add implementation of `SectionWriter` that provides encryption, along with a test utility `SectionWriter` implentation of a data cache.
1. Evaluate and prune `bmt.Hasher` exports wtr `AsyncHasher`

---

This PR does not alter any logic. It merely moves the `AsyncHasher` object to the `file/hasher` package, while exporting a few consts and vars in `bmt/testutil` that are used across the sync and async tests.

### CODE REVIEW

**Review by pradovic — APPROVED** [review]
LGTM, nice work!

File: bmt/testutil/testutil.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package testutil
```
Comment (janos) [inline bmt/testutil/testutil.go:17]: Add copyright header to all new files.

File: file/hasher/hasher_test.go [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+	bmtobj := NewAsyncHasher(ctx, bmth, double, nil)
+	idxs, segments := splitAndShuffle(bmtobj.SectionSize(), data)
+	rand.Shuffle(len(idxs), func(i int, j int) {
+		idxs[i], idxs[j] = idxs[j], idxs[i]
+	})
+
+	t.ReportAllocs()
+	t.ResetTimer()
+	for i := 0; i < t.N; i++ {
+		asyncHash(bmtobj, 0, n, wh, idxs, segments)
```
Comment (janos) [inline file/hasher/hasher_test.go]: Could this benchmark be unexpectedly optimised by compiler as the function result is not used anywhere?

**Review by janos — APPROVED** [review]
LGTM

**Pull request merged**